### DISCUSSION TITLE
Blazor and Mediator

### DISCUSSION THREAD

**Discussion by Gope** [LuckyPennySoftware/MediatR#927] [Q&A]
Hello everybody,
I am coming from the classic Event Aggregator Pattern. While MediatR seems to work more like a ServiceBus (NServiceBus f.e.) where Handlers are autonomous nodes I still can't seem to understand how a Blazor Component can interact with a handler or would the handling Interface (IHandleNotification<foo>) be implemented in the Component like with an EventAggregator. I don't have every Component registered in the DI-Container (and don't want to) but they are mostly built by the Blazor Rendering system. How could a component that is interested in getting a message from any Publisher receive that message? With an EventAggregator one would inject the Aggregator und register but here MediatR seems to register by type and the resolve which may be fine for handlers but not for all components.

Thanks for your help!

**Comment by jbogard** [comment]
MediatR is completely synchronous, it's not really a bus. You shouldn't really compare the two. If you need asynchronous messaging, you should look at something like SignalR.

**Reply by Gope** [reply]
Hello jimmy and thank you so much for taking the time to answer. I don't need an async solution but do not understand how the interaction from an autonomous handler which is created without dependencies via a container is working for UI-Components. How do they play together? How does a UI-Component get a message (via a handler). I've seen such strange and simply lucky solutions using a static event handler. Should I refrain to the classic Event Aggregators?

**Reply by jbogard** [reply]
A UI component has the mediator (event aggregator). The event aggregator has an instance of the container/composition root. That composition root needs to have the notification handlers in its registry to be able to instantiate them. The handlers don't have to be instantiated already, but they need to be registered in the container.

**Comment by richardaubin** [comment]
Implement the Mediatr INotificationHandler<...> on your component class so you can handle the notification event within the component?
